### PR TITLE
HW2 Task03

### PR DESCRIPTION
**Pull request by LinaKrevetochka**

### CODE REVIEW

File: Homework2/Task03/task03.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Example:
+    assert combinations([1, 2], [3, 4]) == [
+        [1, 3],
+        [1, 4],
+        [2, 3],
+        [2, 4],
+    ]
+    """
+    all_comb = []
+    for comb in itertools.product(*args):
```
Comment (Cjkjvfnby) [inline Homework2/Task03/task03.py]: The whole function can be oneliner.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework2/Task03/task03.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    [2, 3],
+    [2, 4],
+]
+"""
+import itertools
+from typing import Any, List
+
+
+def combinations(*args: List[Any]) -> List[List]:
+    return [list(comb) for comb in itertools.product(*args)]
```
Comment (Cjkjvfnby) [inline Homework2/Task03/task03.py]: Can you solve it without itertools?

Comment (LinaKrevetochka) [reply Homework2/Task03/task03.py]: I can do something like this.
```
        def combinations(*args: List[Any]) -> List[List]:
            result = [[]]
            for arg in args:
                result = [x + [y] for x in result for y in arg]
            return result
```

Comment (Cjkjvfnby) [reply Homework2/Task03/task03.py]: Please do something and pass the tests.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
`product` is nice but it kind of cheating lets solve this task without it.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**